mathics/core/atoms.py

```diff
@@ -1098,6 +1104,125 @@ def is_zero(self) -> bool:
 }
 
 
+#
```

Comment:
I propose to split this off (together with a couple minor changes in other files, but not the NumericArray builtin, Part, First, Normal, etc.) into a separate PR - what do you think?

Reply:
Deferring Part, First, Normal, etc. is fine. As you've seen, it's okay to defer implementation when it's made explicit what's needs to be done (and that in of itself will be long and involved). 

I had in the back of my mind to split out the mathics.atom into mathics.atom.numeric, mathics.atom.array, mathics.atom.symbol, and mathics.atom.string (or something like that).  But since that will be another revision here, I suppose I can defer until after NumericArray part 1.

Reply:
OK. So here's what I think should go in Part 1 from this PR:
mathics/core/atoms.py
mathics/core/convert/python.py
mathics/core/keybomparable.py
the first section of test/builtin/test_numericarray.py,  which relates to NumericArray atom as Python object

How would you like to proceed?
* I prepare a PR with those changes and submit upstream and we all review there?
* I prepare a PR with those changes and submit against my fork (like this PR) for you to review first?
* You first review just those files in this PR then we can submit as PR upstream?
* You mull it over for a while and we discuss on Tuesday?

Reply:
> OK. So here's what I think should go in Part 1 from this PR: mathics/core/atoms.py mathics/core/convert/python.py mathics/core/keybomparable.py the first section of test/builtin/test_numericarray.py, which relates to NumericArray atom as Python object
> 

Great! 
> How would you like to proceed?
> 
> * I prepare a PR with those `hanges` and submit upstream and we all review there?

Perfectly fine. 

> * I prepare a PR with those changes and submit against my fork (like this PR) for you to review first?

Up to you.

> * You first review just those files in this PR then we can submit as PR upstream?

Up to you.

> * You mull it over for a while and we discuss on Tuesday?

Will probably do in any event.